libraries/math/src/vector3/coreHelpers.ts

```diff
@@ -159,6 +159,67 @@ export class Vector3Utils {
         const tb = Math.sin(t * theta) / sinTheta;
         return Vector3Utils.add(Vector3Utils.scale(a, ta), Vector3Utils.scale(b, tb));
     }
+    /**
+     * multiply
+     *
+     * Element-wise multiplication of two vectors together.
+     * Not to be confused with {@link Vector3Utils.dot} product or {@link Vector3Utils.cross} product
+     */
+    static multiply(a: Vector3, b: Vector3): Vector3 {
+        return {
+            x: a.x * b.x,
+            y: a.y * b.y,
+            z: a.z * b.z,
+        };
+    }
+
+    /**
+     * rotateX
+     *
+     * Rotates the vector around the x axis counterclockwise (left hand rule)
+     * @param a - Angle in radians
+     */
+    static rotateX(v: Vector3, a: number): Vector3 {
+        let cos = Math.cos(a);
+        let sin = Math.sin(a);
+        return {
+            x: v.x,
+            y: v.y * cos - v.z * sin,
+            z: v.z * cos + v.y * sin,
+        };
+    }
+
+    /**
+     * rotateY
+     *
+     * Rotates the vector around the y axis counterclockwise (left hand rule)
+     * @param a - Angle in radians
+     */
+    static rotateY(v: Vector3, a: number): Vector3 {
+        let cos = Math.cos(a);
+        let sin = Math.sin(a);
+        return {
+            x: v.x * cos - v.z * sin,
+            y: v.y,
+            z: v.z * cos + v.x * sin,
+        };
+    }
```

Comment:
I believe you inadvertently inverted this one too. rotateX and rotateZ are now properly counter clockwise rotation, but rotateY is now a clockwise rotation, and your unit test confirms this.

This should be 
```suggestion
    static rotateY(v: Vector3, a: number): Vector3 {
        let cos = Math.cos(a);
        let sin = Math.sin(a);
        return {
            x: v.x * cos + v.z * sin,
            y: v.y,
            z: v.z * cos - v.x * sin,
        };
    }
```

Reply:
Oh I thought in the other conversation you meant all of them were the wrong direction.

I guess in hindsight the test does very clearly tell which direction it is spinning. I guess I just saw that the angle was correct and called it a day.